improver_tests/calibration/reliability_calibration/test_ConstructRealizationCalibrationTables.py

```diff
@@ -0,0 +1,366 @@
+        forecast cube does not contain all the required metadata to copy."""
+
+        diagnostic = self.forecast_1.coord(var_name='threshold').name()
+        result = Plugin._define_metadata(self.forecast_1, diagnostic)
+
+        self.assertIsInstance(result, dict)
+        self.assertEqual(result, self.expected_attributes)
+
+
+class Test__create_reliability_table_cube(Test_Setup):
+
+    """Test the _create_reliability_table_cube method."""
+
+    def test_valid_inputs(self):
+        """Test the cube returned has the structure expected."""
+
+        input_data = np.ones((3, 5, 3, 3))
+        forecast_slice = next(self.forecast_1.slices_over('air_temperature'))
+        result = Plugin()._create_reliability_table_cube(input_data,
+                                                         forecast_slice)
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertSequenceEqual(result.shape, input_data.shape)
+        self.assertEqual(result.name(), "reliability_calibration_table")
+        self.assertEqual(result.attributes, self.expected_attributes)
+
+    def test_invalid_inputs(self):
+        """Test that an exception is raised if the input data is not of the
+        expected dimensions."""
+
+        input_data = np.ones((3, 4, 3, 3))
+        forecast_slice = next(self.forecast_1.slices_over('air_temperature'))
+
+        msg = ('The reliability table data does not have the expected '
+               'dimensions.')
+        with self.assertRaisesRegex(ValueError, msg):
+            Plugin()._create_reliability_table_cube(input_data,
+                                                    forecast_slice)
+
+
+class Test__populate_reliability_bins(Test_Setup):
+
+    """Test the _populate_reliability_bins method."""
+
+    def test_table_values(self):
+        """Test the reliability table returned has the expected values for the
+        given inputs."""
+
+        forecast_slice = next(self.forecast_1.slices_over('air_temperature'))
+        truth_slice = next(self.truth_1.slices_over('air_temperature'))
+        result = Plugin()._populate_reliability_bins(
+            forecast_slice.data, truth_slice.data)
+
+        self.assertSequenceEqual(result.shape, self.expected_table_shape)
+        assert_array_equal(result, self.expected_table)
+
+
+class Test_process(Test_Setup):
+
+    """Test the process method."""
+
+    def test_return_type(self):
+        """Test the process method returns a reliability table cube."""
+
+        result = Plugin().process(self.forecasts, self.truths)
+
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertEqual(result[0].name(), "reliability_calibration_table")
+
+    def test_table_values(self):
+        """Test that cube values are as expected when process has sliced the
+        inputs up for processing and then summed the contributions from the
+        two dates. Note that the table tested here is still for only one
```

Comment:
Is this true `still for only one threshold (283K)`, as it seems like the input forecast has two thresholds?

Reply:
Hopefully this has been clarified in the doc-string.